scripts/regression/test_runner.py

```diff
@@ -151,32 +161,70 @@ class BenchmarkResult:
         ):
             regression_list.append(BenchmarkResult(benchmark, old_res, new_res))
         else:
-            other_results.append(BenchmarkResult(benchmark, old_res, new_res))
+            try:
+                improved_by = int((float(old_res) - float(new_res)) * 100.0 / float(old_res))
+            except Exception:
+                improved_by = 0
+            if improved_by > 0:
+                improvements_list.append(BenchmarkResult(benchmark, old_res, new_res, improvement=improved_by))
+            else:
+                other_results.append(BenchmarkResult(benchmark, old_res, new_res))
     benchmark_list = [res.benchmark for res in regression_list]
 
+time_old = geomean(old_runner.complete_timings)
+time_new = geomean(new_runner.complete_timings)
+
 exit_code = 0
 regression_list.extend(error_list)
 summary = []
 if len(regression_list) > 0:
-    exit_code = 1
-    print(
-        '''====================================================
-==============  REGRESSIONS DETECTED   =============
+    if not (isinstance(time_old, str) or isinstance(time_new, str)):
+        regression_percentage = int((time_new - time_old) * 100.0 / time_old)
+    else:
+        regression_percentage = 0
+    if (
+        isinstance(MAX_ALLOWED_REGRESS_PERCENTAGE, int)
+        and isinstance(regression_percentage, int)
+        and regression_percentage < MAX_ALLOWED_REGRESS_PERCENTAGE
+    ):
+        # allow individual regressions less than 10% when overall geomean had improved or hadn't change (on large benchmarks)
```

Comment:
I'm not seeing the code where we go through the regressions_list to check each regression for less than 10%. Instead we are only checking the geomean. 

Am I missing the code for the individual checks somewhere?